apis/apps/v1alpha1/cluster_types.go

```diff
@@ -652,6 +652,16 @@ type ClusterComponentSpec struct {
 	// +optional
 	EnabledLogs []string `json:"enabledLogs,omitempty"`
 
+	// Specifies Annotations to override or add for underlying Pods.
+	//
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// List of environment variables to add.
+	//
+	// +optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
```

Comment:
The API documentation should explain the relationship between the environment variables defined here and other environment variables (such as those in the podSpec of cmpd and vars), including the mechanism by which they take effect. Here is a related user request: https://github.com/apecloud/kubeblocks/discussions/7502

Reply:
done

Reply:
here is the feature request for user-provided env: https://github.com/apecloud/kubeblocks/discussions/7502